aptos-move/framework/aptos-stdlib/sources/simple_map.move

```diff
@@ -77,6 +77,24 @@ module aptos_std::simple_map {
         vector::push_back(&mut map.data, Element { key, value });
     }
 
+    public fun upsert<Key: store + drop + copy, Value: store + drop + copy>(
+        map: &mut SimpleMap<Key, Value>,
+        key: &Key,
+        value: &Value
+    ) {
+        let len = std::vector::length(&map.data);
+        let i = 0;
+        while (i < len) {
+            let element = vector::borrow_mut(&mut map.data, i);
+            if (&element.key == key) {
+                element.value = *value;
+                return
+            };
+            i = i + 1;
+        };
+        vector::push_back(&mut map.data, Element { key: *key, value: *value });
```

Comment:
Shouldn't this only be when the entry doesn't exist?

Reply:
Yes and it is otherwise it returns when key is found. Note that now it's not inline function and hence we can return